Lib/test/test_platform.py

```diff
@@ -353,5 +353,22 @@ def test_macos(self):
                     self.assertEqual(platform.platform(), expected)
 
 
+    def test_non_ascii_digits_windows_version(self):
+        # Non-Ascii digits must be rejected in version numbers.
+        version_strings = {
+            'Microsoft Windows [Version 6.1.7601]': ('Microsoft', 'Windows', '6.1.7601'),
+            'Microsoft Windows [Version ٦.1.7601]': None, # East-arabic numerals
+            'Microsoft Windows [Version 6.１.7601]': None, # Full-width numerals
+            'Microsoft Windows [Version 6.1.760１]': None, # Full-width numerals
+        }
+        for version_string, expected_groups in version_strings.items():
+            if match := platform._ver_output.match(version_string):
+                self.assertNotEqual(expected_groups, None)
+                self.assertEqual(expected_groups, match.groups())
+            else:
+                self.assertEqual(expected_groups, None)
```

Comment:
You should use `assertIsNone` and `assertIsNotNone` instead of `assertEqual`.  If the test is refactored you might be able to use `assertRegex` and `assertNotRegex` too.